wasm2c/examples/fac/fac.c

```diff
@@ -485,58 +485,96 @@ static inline void memory_init(wasm_rt_memory_t* dest,
 
 typedef struct {
   uint32_t func_type_index;
-  wasm_rt_funcref_t func;
+  wasm_rt_function_ptr_t func;
   size_t module_offset;
 } wasm_elem_segment_expr_t;
```

Comment:
It's definitely possible! The `module_offset` field here is what handles that situation (which I believe is tested in the testsuite, e.g. in linking.wast). FWIW this code was added already in handling bulk memory ops.

The reason these are separate types is that a `wasm_rt_funcref_t` includes a pointer to an instance of the originating module, which is what you want when calling a funcref. But when CWriter is representing the elem expr, we don't have the instance of the originating module because the module hasn't been instantiated yet. So instead the elem segment expr contains basically a pointer-to-member; the offset within the module instance of where the function's originating module instance will be found, so that `funcref_table_init` (previously just `table_init`) can properly initialize the table at runtime. Also, the `wasm_elem_segment_expr_t` stores the function's type index within the module, whereas a `wasm_rt_funcref_t` stores the global opaque type ID returned by `wasm_rt_register_func_type`.